.travis.yml

```diff
@@ -2,17 +2,27 @@ language: go
 go:
   - "1.9.4"
   - "1.10"
+cache:
+  directories:
+    - vendor/
```

Comment:
might we run into issues with this due to the inconsistencies we've been experiencing with `dep`? If the build breaks, is there any way for us to remove `vendor` manually?

Another option may be to cache `$GOPATH`, that way all of the packages are already installed (or some recent version), but the `vendor` directory is still fresh. This should cause it to check out and copy packages that are unchanged, and only fetch diffs for dependencies that have updated.

Reply:
You can always manually delete the cache from travis-ci.org.

I added caching of the $GOCACHE dir (only for go 1.10+) which speeds up building, and also caches tests results! This means that unit tests run much faster now.

Regarding caching the $GOPATH, you cannot do that entirely, as the lnd directory will conflict. I did add caching of a few of the deps , however, as it sped up some of the steps (make lint, btcd, dep). Caching more dirs can be considered, but it doesn't seem like there's much more to gain there (I modified travis.yml to do each step at a time, such that we can see how long it takes).

Reply:
We'll be switching over to go modules soon after this is merged, which'll eliminate all the issues we're seeing with dep.